### PR TITLE
NETOBSERV-1575: Upgrade to node18, bump some libs

### PR DESCRIPTION
**Pull request by jotak**
## Description

Upgrade to node18, bump some libs

## Dependencies

n/a

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [x] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [ ] Does this PR require a product release notes entry?
  * [ ] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [ ] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

**Comment by jotak** [comment]
@memodi @jpinsonneau I don't see any regression while using node18 for build, but I haven't checked yet the cypress tests, we'll see the job results

When building on my machine, I had to increase the open files limit on my system (`ulimit -n 65535`) ; perhaps you'll need to do it as well if you build locally

File: Dockerfile [hunk 1]
```diff
@@ -3,7 +3,13 @@ ARG BUILDSCRIPT=
 ARG LDFLAGS=
 ARG TARGETPLATFORM=linux/amd64
 ARG BUILDPLATFORM=linux/amd64
-FROM --platform=$BUILDPLATFORM docker.io/library/node:16-alpine as web-builder
+FROM --platform=$BUILDPLATFORM docker.io/library/node:18-alpine as web-builder
+
+# npm cache stuff below help get rid of an error "EMFILE, too many open files"
+RUN npm cache clean --force
+RUN npm install -g npm@10.2.4
+RUN npm cache verify
```
Comment (jotak) [inline Dockerfile:11]: actually this perhaps isn't useful, i'll retry without it

**Comment by jotak** [comment]
/hold

I keep getting randomly the `EMFILE, too many open files` ; ideally I'd prefer to no have it without changing my system defaults, not sure if there's a way to avoid it

**Review by jpinsonneau — APPROVED** [review]
LGTM, build, start and images works fine

**Comment by memodi** [comment]
> @memodi @Amoghrd although building with node16 will still be possible, perhaps you will want to align the prow builds to use node18 as well?

@jotak For node.js version in prow builds we only rely Dockerfile and Dockerfile.cypress changes so should be good. I see both [prow/images](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/netobserv_network-observability-console-plugin/505/pull-ci-netobserv-network-observability-console-plugin-main-images/1771188194289651712) and [prow/plugin-cypress](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/netobserv_network-observability-console-plugin/505/pull-ci-netobserv-network-observability-console-plugin-main-plugin-cypress/1771188194327400448) has passed on this PR, so this should be good to merge. Thanks!

**Comment by Amoghrd** [comment]
Will run the automated backend regression to see if there's any issue

**Comment by Amoghrd** [comment]
Oh I misread it. Need to run frontend tests for this PR. I can only test frontend tests on downstream version of operator.
@jotak You can go ahead and merge it, I will verify it post-merge by running the automated frontend tests

**Comment by jotak** [comment]
thanks @memodi @Amoghrd 
/approve

**Pull request merged**